week-06/assessment/field-agent/src/main/java/learn/field_agent/data/SecurityClearanceJdbcTemplateRepository.java

```diff
@@ -25,4 +31,52 @@ public SecurityClearance findById(int securityClearanceId) {
                 .stream()
                 .findFirst().orElse(null);
     }
+    public SecurityClearance findByName(String name) {
+        final String sql = "SELECT security_clearance_id, name security_clearance_name FROM security_clearance WHERE name = ?;";
+        return jdbcTemplate.query(sql, new SecurityClearanceMapper(), name).stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public List<SecurityClearance> findAll() {
+        final String sql = "SELECT security_clearance_id, name security_clearance_name FROM security_clearance limit 1000";
+        return jdbcTemplate.query(sql, new SecurityClearanceMapper());
+    }
+
+    @Override
+    public SecurityClearance add(SecurityClearance securityClearance) {
+        final String sql = "INSERT INTO security_clearance (`name`) VALUES (?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        // literally just following convention of the code in the project,
+
+        int rowsAffected = jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            ps.setString(1, securityClearance.getName());
+            return ps;
+        }, keyHolder);
+
+        if (rowsAffected <= 0) {
+            return null;
+        }
+
+        securityClearance.setSecurityClearanceId(keyHolder.getKey().intValue());
+        return securityClearance;
+    }
+
+    @Override
+    public boolean update(SecurityClearance securityClearance) {
+        final String sql = "UPDATE security_clearance SET `name` = ? WHERE security_clearance_id = ?";
+        return jdbcTemplate.update(sql, securityClearance.getName(), securityClearance.getSecurityClearanceId()) > 0;
+    }
+
+    @Override
+    public boolean deleteById(int securityClearanceId) {
+        final String deleteAgentSecurityClearance = "DELETE FROM agency_agent WHERE security_clearance_id = ?";
```

Comment:
I like the cascading delete. It's an appropriate strategy.
Consider annotating with `@Transactional` since that will create a database transaction _before_ the method call and commit or reject it _after_ the method call.